### DISCUSSION TITLE
Pattern for Highlighting Books

### DISCUSSION THREAD

**Discussion by rafaeltgt** [danielmiessler/Fabric#580] [Ideas]
Hello everyone, 

I noticed that there are two patterns for books, but they ask the LLM to bring content it knows about the book already. I wanted something more robust, and I thought about providing the actual content of a book along with a prompt to extract information (due to the context window, I am doing one chapter at a time).

The extract_wisdom pattern worked well. But I needed a more specialized prompt. 

MAIN ISSUE: Some books have a lot of filler, where the author takes a long time to convey an idea. Therefore, I thought of a prompt for the LLM to highlight the main parts of the text. 

I'm not saying all writers do this or that we should do this with all books. But I think it's valid for some situations:

- Getting to know a book better before committing to reading it in full
- Revisiting the ideas of a book you have already read
- Creating a mind map of the book
- Your reading list is growing uncontrollably

Would you add any additional instructions? Comment below.

PROMPT:
{CHAPTER CONTENT}

Act as a textual analysis expert. Your task is to analyze the provided text and identify the core ideas and the quotes that support these ideas. Divide your response into two sections: CORE IDEAS and QUOTES.

**CORE IDEAS:**
List the main ideas of the text concisely and comprehensively. Only consider the ideas present in the text, without adding external knowledge. Ensure that all key ideas from the text are included.

**QUOTES:**
For each idea listed in CORE IDEAS, provide direct quotes from the text that support these ideas. Use the exact text from the input. Ensure that the quotes are accurate and clearly correspond to the listed ideas.

Example response structure:
```
### CORE IDEAS:
1. [Main idea 1]
2. [Main idea 2]
3. [Main idea 3]

### QUOTES:
1. [Main idea 1]
   - "Exact quote from the text."
2. [Main idea 2]
   - "Exact quote from the text."
3. [Main idea 3]
   - "Exact quote from the text."
```

OPTIONAL:**Step-by-step exercises or proposed actions**: If the text includes any exercises or proposed actions, list the step-by-step instructions for each action as proposed in the book. Do not include any actions outside the selected excerpt, even if you know the content of the book.

**Comment by adamgroenhout** [comment]
Have you considered just modifying the extract_book_ideas pattern to (1) list core ideas (which would be supplemented by the existing 50-100 ideas), (2) include/list quotes, and (3) list the exercises/actions (which btw, I think is a cool idea)... ?
